src/kernels/composable_kernel/include/utility/reduction_common.hpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef CK_REDUCTION_COMMON_HPP
+#define CK_REDUCTION_COMMON_HPP
+
+#include "float_type.hpp"
+
+// this enumerate should be synchronized with include/miopen/reduce_common.hpp
+namespace ck {
+typedef enum {
+    Reduce_DirectThreadWise = 1,
+    Reduce_DirectWarpWise   = 2,
+    Reduce_BlockWise        = 3,
+    Reduce_MultiBlock       = 4
+} ckReductionMethod_t; // end of namespace ck
+
+typedef enum {
+    CK_REDUCE_TENSOR_ADD = 0,
+    CK_REDUCE_TENSOR_MUL = 1,
+    CK_REDUCE_TENSOR_MIN = 2,
+    CK_REDUCE_TENSOR_MAX = 3,
+    // CK_REDUCE_TENSOR_AMAX = 4,
+    // CK_REDUCE_TENSOR_AVG =  5,
+    // CK_REDUCE_TENSOR_NORM1 = 6,
+    // CK_REDUCE_TENSOR_NORM2 = 7,
+    // CK_REDUCE_TENSOR_MUL_NO_ZEROS = 8,
+} ckReduceTensorOp_t;
```

Comment:
Also remove ```CK``` and ```ck``` from here, and from following codes

Reply:
Agree !

Reply:
Check https://github.com/ROCmSoftwarePlatform/MIOpen/pull/322/commits/63eccb3a606d9d30b2c08fce8b7f42f72aecb40e